service/history/replicatorQueueProcessor.go

```diff
@@ -467,6 +468,7 @@ func (p *replicatorQueueProcessorImpl) getTasks(
 		readLevel = taskInfo.GetTaskID()
 		if replicationTask != nil {
 			replicationTasks = append(replicationTasks, replicationTask)
+			time.Sleep(p.taskGenerationWait())
```

Comment:
I am not sure how much timer scheduling resource it will cost on the CPU. From that point, I would prefer using a ratelimiter.

Also seems like it will try to load dynamic config every 10 milliseconds, that will also cause lots of GC activity. Probably use the same value across a single getTasks() call.

Reply:
If there is concern on CPU, I updated to a rate limiter.